### ISSUE TITLE
com.hazelcast.jet.sql_slow.SqlOrderBySlowTest

### ISSUE THREAD

**Issue by bpoplawski** [hazelcast/hazelcast#20855]
5.0.z

Note that similar issue was probably fixed on 5.1 but we expect it also on 5.0.z
https://github.com/hazelcast/hazelcast/issues/19884

Failed on OracleJDK8:  https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-5.maintenance-OracleJDK8-nightly/23/testReport/com.hazelcast.jet.sql_slow/SqlOrderBySlowTest/com_hazelcast_jet_sql_slow_SqlOrderBySlowTest/

<details><summary>Stacktrace:</summary>

```
java.lang.IllegalStateException: Instances haven't been shut down: [HazelcastInstance{name='adoring_torvalds', node=[127.0.0.1]:5701}, HazelcastInstance{name='quirky_torvalds', node=[127.0.0.1]:5702}]
	at com.hazelcast.test.AfterClassesStatement.evaluate(AfterClassesStatement.java:46)
	at com.hazelcast.test.OverridePropertyRule$1.evaluate(OverridePropertyRule.java:66)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

</details>

<details><summary>Standard output:</summary>

```
Finished Running Test: test_indexScan in 1200.932 seconds.
22:49:16,330  INFO |test_indexScan| - [InternalPartitionService] hz.quirky_torvalds.migration - [127.0.0.1]:5702 [dev] [5.0.3-SNAPSHOT] Fetching partition tables from cluster to determine the most recent one... Local stamp: 2512163267035406600
22:49:16,330  INFO |test_indexScan| - [InternalPartitionService] hz.quirky_torvalds.migration - [127.0.0.1]:5702 [dev] [5.0.3-SNAPSHOT] Most recent partition table is determined.
22:49:16,330  INFO |test_indexScan| - [InternalPartitionService] hz.quirky_torvalds.migration - [127.0.0.1]:5702 [dev] [5.0.3-SNAPSHOT] Applying the most recent of partition state...
22:49:16,330  INFO || - [Node] Time-limited test - [127.0.0.1]:5702 [dev] [5.0.3-SNAPSHOT] Shutting down connection manager...
22:49:16,330  WARN || - [TestClientRegistry$MockedServerConnection] Time-limited test - Server connection closed: null
22:49:16,330  INFO || - [MockServer] Time-limited test - [127.0.0.1]:5702 [dev] [5.0.3-SNAPSHOT] Removed connection to endpoint: [127.0.0.1]:40006, connection: MockedNodeConnection{ remoteAddress = [127.0.0.1]:40006, localAddress = [127.0.0.1]:5702, connectionId = 3}
22:49:16,330  WARN || - [TestClientRegistry$MockedServerConnection] Time-limited test - Server connection closed: null
22:49:16,331  INFO || - [MockServer] Time-limited test - [127.0.0.1]:5702 [dev] [5.0.3-SNAPSHOT] Removed connection to endpoint: [127.0.0.1]:40005, connection: MockedNodeConnection{ remoteAddress = [127.0.0.1]:40005, localAddress = [127.0.0.1]:5702, connectionId = 3}
22:49:16,331  INFO || - [Node] Time-limited test - [127.0.0.1]:5702 [dev] [5.0.3-SNAPSHOT] Shutting down node engine...
22:49:16,331  WARN |test_indexScan| - [TcpClientConnection] pool-10-thread-1 - hz.client_1 [dev] [5.0.3-SNAPSHOT] MockedClientConnection{localAddress=[127.0.0.1]:40006, super=ClientConnection{alive=false, connectionId=3, channel=null, remoteAddress=[127.0.0.1]:5702, lastReadTime=2022-02-28 22:49:16.178, lastWriteTime=2022-02-28 22:49:08.481, closedTime=2022-02-28 22:49:16.330, connected server version=5.0.3-SNAPSHOT}} closed. Reason: com.hazelcast.spi.exception.TargetDisconnectedException[Mocked Remote socket closed]
com.hazelcast.spi.exception.TargetDisconnectedException: Mocked Remote socket closed
	at com.hazelcast.client.test.TestClientRegistry$MockedTcpClientConnection$4.run(TestClientRegistry.java:307) [hazelcast-5.0.3-SNAPSHOT-tests.jar:5.0.3-SNAPSHOT]
	at com.hazelcast.client.test.TwoWayBlockableExecutor$BlockableRunnable.run(TwoWayBlockableExecutor.java:98) [hazelcast-5.0.3-SNAPSHOT-tests.jar:5.0.3-SNAPSHOT]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_311]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_311]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_311]
22:49:16,331  WARN |test_indexScan| - [TcpClientConnection] pool-11-thread-1 - hz.client_2 [dev] [5.0.3-SNAPSHOT] MockedClientConnection{localAddress=[127.0.0.1]:40005, super=ClientConnection{alive=false, connectionId=3, channel=null, remoteAddress=[127.0.0.1]:5702, lastReadTime=2022-02-28 22:49:16.178, lastWriteTime=2022-02-28 22:49:09.531, closedTime=2022-02-28 22:49:16.331, connected server version=5.0.3-SNAPSHOT}} closed. Reason: com.hazelcast.spi.exception.TargetDisconnectedException[Mocked Remote socket closed]
com.hazelcast.spi.exception.TargetDisconnectedException: Mocked Remote socket closed
	at com.hazelcast.client.test.TestClientRegistry$MockedTcpClientConnection$4.run(TestClientRegistry.java:307) [hazelcast-5.0.3-SNAPSHOT-tests.jar:5.0.3-SNAPSHOT]
	at com.hazelcast.client.test.TwoWayBlockableExecutor$BlockableRunnable.run(TwoWayBlockableExecutor.java:98) [hazelcast-5.0.3-SNAPSHOT-tests.jar:5.0.3-SNAPSHOT]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_311]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_311]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_311]
22:49:16,331  WARN |test_indexScan| - [TwoWayBlockableExecutor] pool-10-thread-1 - Dropping outgoing runnable since other end closed. Client Closed EOF. MockedClientConnection{localAddress=[127.0.0.1]:40006, super=ClientConnection{alive=false, connectionId=3, channel=null, remoteAddress=[127.0.0.1]:5702, lastReadTime=2022-02-28 22:49:16.178, lastWriteTime=2022-02-28 22:49:08.481, closedTime=2022-02-28 22:49:16.330, connected server version=5.0.3-SNAPSHOT}}
22:49:16,331  INFO |test_indexScan| - [ClientConnectionManager] pool-10-thread-1 - hz.client_1 [dev] [5.0.3-SNAPSHOT] Removed connection to endpoint: [127.0.0.1]:5702:3d393d45-e9de-4a7b-a604-93eee5812981, connection: MockedClientConnection{localAddress=[127.0.0.1]:40006, super=ClientConnection{alive=false, connectionId=3, channel=null, remoteAddress=[127.0.0.1]:5702, lastReadTime=2022-02-28 22:49:16.178, lastWriteTime=2022-02-28 22:49:08.481, closedTime=2022-02-28 22:49:16.330, connected server version=5.0.3-SNAPSHOT}}
22:49:16,331  WARN |test_indexScan| - [TwoWayBlockableExecutor] pool-11-thread-1 - Dropping outgoing runnable since other end closed. Client Closed EOF. MockedClientConnection{localAddress=[127.0.0.1]:40005, super=ClientConnection{alive=false, connectionId=3, channel=null, remoteAddress=[127.0.0.1]:5702, lastReadTime=2022-02-28 22:49:16.178, lastWriteTime=2022-02-28 22:49:09.531, closedTime=2022-02-28 22:49:16.331, connected server version=5.0.3-SNAPSHOT}}
22:49:16,331  INFO |test_indexScan| - [ClientConnectionManager] pool-11-thread-1 - hz.client_2 [dev] [5.0.3-SNAPSHOT] Removed connection to endpoint: [127.0.0.1]:5702:3d393d45-e9de-4a7b-a604-93eee5812981, connection: MockedClientConnection{localAddress=[127.0.0.1]:40005, super=ClientConnection{alive=false, connectionId=3, channel=null, remoteAddress=[127.0.0.1]:5702, lastReadTime=2022-02-28 22:49:16.178, lastWriteTime=2022-02-28 22:49:09.531, closedTime=2022-02-28 22:49:16.331, connected server version=5.0.3-SNAPSHOT}}
22:49:16,331  INFO |test_indexScan| - [LifecycleService] pool-11-thread-1 - hz.client_2 [dev] [5.0.3-SNAPSHOT] HazelcastClient 5.0.3-SNAPSHOT (20220228 - 3b5559b) is CLIENT_DISCONNECTED
22:49:16,331  INFO |test_indexScan| - [LifecycleService] pool-10-thread-1 - hz.client_1 [dev] [5.0.3-SNAPSHOT] HazelcastClient 5.0.3-SNAPSHOT (20220228 - 3b5559b) is CLIENT_DISCONNECTED
22:49:16,332  INFO |test_indexScan| - [ClientConnectionManager] hz.client_2.internal-8 - hz.client_2 [dev] [5.0.3-SNAPSHOT] Trying to connect to cluster: dev
22:49:16,332  INFO |test_indexScan| - [ClientConnectionManager] hz.client_1.internal-8 - hz.client_1 [dev] [5.0.3-SNAPSHOT] Trying to connect to cluster: dev
22:49:16,332  INFO |test_indexScan| - [ClientConnectionManager] hz.client_2.internal-8 - hz.client_2 [dev] [5.0.3-SNAPSHOT] Trying to connect to Member [127.0.0.1]:5702 - 3d393d45-e9de-4a7b-a604-93eee5812981
22:49:16,332  INFO |test_indexScan| - [ClientConnectionManager] hz.client_1.internal-8 - hz.client_1 [dev] [5.0.3-SNAPSHOT] Trying to connect to Member [127.0.0.1]:5702 - 3d393d45-e9de-4a7b-a604-93eee5812981
22:49:16,332  WARN |test_indexScan| - [ClientConnectionManager] hz.client_1.internal-8 - hz.client_1 [dev] [5.0.3-SNAPSHOT] Exception during initial connection to Member [127.0.0.1]:5702 - 3d393d45-e9de-4a7b-a604-93eee5812981: com.hazelcast.core.HazelcastException: java.io.IOException: Can not connect to [127.0.0.1]:5702: instance does not exist
22:49:16,332  WARN |test_indexScan| - [ClientConnectionManager] hz.client_2.internal-8 - hz.client_2 [dev] [5.0.3-SNAPSHOT] Exception during initial connection to Member [127.0.0.1]:5702 - 3d393d45-e9de-4a7b-a604-93eee5812981: com.hazelcast.core.HazelcastException: java.io.IOException: Can not connect to [127.0.0.1]:5702: instance does not exist
22:49:16,332  INFO |test_indexScan| - [ClientConnectionManager] hz.client_2.internal-8 - hz.client_2 [dev] [5.0.3-SNAPSHOT] Trying to connect to [127.0.0.1]:5701
22:49:16,332  INFO |test_indexScan| - [ClientConnectionManager] hz.client_1.internal-8 - hz.client_1 [dev] [5.0.3-SNAPSHOT] Trying to connect to [127.0.0.1]:5701
22:49:16,332  WARN |test_indexScan| - [ClientConnectionManager] hz.client_2.internal-8 - hz.client_2 [dev] [5.0.3-SNAPSHOT] Exception during initial connection to [127.0.0.1]:5701: com.hazelcast.core.HazelcastException: java.io.IOException: Can not connect to [127.0.0.1]:5701: instance does not exist
22:49:16,332  WARN |test_indexScan| - [ClientConnectionManager] hz.client_1.internal-8 - hz.client_1 [dev] [5.0.3-SNAPSHOT] Exception during initial connection to [127.0.0.1]:5701: com.hazelcast.core.HazelcastException: java.io.IOException: Can not connect to [127.0.0.1]:5701: instance does not exist
22:49:16,333  WARN |test_indexScan| - [ClientConnectionManager] hz.client_1.internal-8 - hz.client_1 [dev] [5.0.3-SNAPSHOT] Unable to get live cluster connection, retry in 1000 ms, attempt: 1, cluster connect timeout: INFINITE, max backoff: 30000 ms
22:49:16,333  WARN |test_indexScan| - [ClientConnectionManager] hz.client_2.internal-8 - hz.client_2 [dev] [5.0.3-SNAPSHOT] Unable to get live cluster connection, retry in 1000 ms, attempt: 1, cluster connect timeout: INFINITE, max backoff: 30000 ms
22:49:16,334  INFO || - [NodeExtension] Time-limited test - [127.0.0.1]:5702 [dev] [5.0.3-SNAPSHOT] Destroying node NodeExtension.
22:49:16,334  INFO || - [Node] Time-limited test - [127.0.0.1]:5702 [dev] [5.0.3-SNAPSHOT] Hazelcast Shutdown is completed in 251 ms.
22:49:16,334  INFO || - [LifecycleService] Time-limited test - [127.0.0.1]:5702 [dev] [5.0.3-SNAPSHOT] [127.0.0.1]:5702 is SHUTDOWN
```

</details>

**Comment by viliam-durina** [comment]
The fix for #19884 is merged in `5.0.z`: https://github.com/hazelcast/hazelcast/commit/2bb01b82c684e8b568df70d0b19a4630a32116b7. This must be some other cause then...

**Comment by viliam-durina** [comment]
I'm closing this because there are no further occurrences logged. It looks like a test issue anyway.
